### ISSUE TITLE
A property that is updated in a batch is not rendered correctly.

### ISSUE THREAD

**Issue by ilyavf** [canjs/canjs#2191]
Simplified example: http://jsbin.com/cicazo/edit?js,output
Bit-tabs example: http://jsbin.com/jiluqi/edit?html,js,output

I ran into this using bit-tabs when tried to show bit-tabs conditionally. The panels.length would stay 0 when two panels present, and this is causing the component not being displayed correctly (panels are missing). The viewModel's values are OK.

If a conditional property is changed **inside a batch** then stache does not show correctly non-primitive properties **inside a child component fragment** if the change happens inside the component's inserted event. This happens for **can-2.3.8 and can-2.3.9**.

```html
{{#if isVisible}}
    {{myMap.test}}
    <my-component>
        {{myMap.test}}
    </my-component>
{{/if}}
```
is rendered as:
```html
new-value
<my-component>
    old-value
</my-component>
```
when
```javascript
var vm = {
    isVisible: false,
    myMap: {test: 'old-value'}  // primitive values (direct vm properties) are OK.
};
var myComponent = {
    events: {
        inserted: function(){
            vm.attr('myMap.test', 'new-value');
        }
    }
}
can.batch.start();             // <<< without a batch this works correctly.
vm.attr('isVisible', true);
can.batch.stop();
```

**Comment by justinbmeyer** [comment]
please try with 2.3.10 and let us know if this is still breaking.

**Comment by justinbmeyer** [comment]
seems similar to #2186

**Comment by ilyavf** [comment]
The test is still breaking in 2.3.10.

**Comment by ilyavf** [comment]
The test requires a component being inserted into DOM to fire the inserted event.

**Comment by ilyavf** [comment]
A scenario without a batch: if the visibility property has a setter then stache does not update the template:

http://jsbin.com/fekava/edit?js,output

```javascript
var vm = can.Map.extend({
    define: {
        isVisible: {
            value: false,
            set: function(val){ return val; } //<<< without a setter it works fine.
        },
        myMap: {
            value: {test: 'old-value'}        // primitive values are OK.
        }
    },
};
var myComponent = {
    events: {
        inserted: function(){
            vm.attr('myMap.test', 'new-value');
        }
    }
};
vm.attr('isVisible', true);

{{#if isVisible}}
    {{myMap.test}}                 {{! 'new-value' will be rendered here }}
    <my-component>
        {{myMap.test}}             {{! <<< 'old-value' will be rendered here }}
    </my-component>
{{/if}}
```

**Comment by dylanrtt** [comment]
> A scenario without a batch: if the visibility property has a setter then stache does not update the template:

I think that's because the presence of a setter [adds a batch](https://github.com/canjs/canjs/blob/master/map/define/define.js#L143), so it's still a batching issue.

**Comment by nlundquist** [comment]
This has been resolved as part of the fixes for #2198 and #2203.

**Comment by daffl** [comment]
I'm going to close this. Please reopen if it still persists.
